<launch>
  <node pkg="tf" type="static_transform_publisher" name="laser_to_base" 
    args="-0.0 0 0.0 0 0 0 base_link laser 50" />
  <node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
    <rosparam>
    maxUrange: 4.0
    maxRange: 4.0
    lstep: 0.05
    iterations: 5
    xmin: -20.0
    ymin: -20.0
    xmax: 20.0
    ymax: 20.0
    srr: 0.1
    stt: 0.2
    str: 0.1
    delta: 0.1
    linearUpdate: 1.0
    angularUpdate: 0.05
    particles: 80
    temporalUpdate: -1.0
    map_update_interval: 5.0
    minimumScore: 0.0
    </rosparam>
  </node>
  <node pkg="move_base" type="move_base" respawn="false" name="move_base_node" output="screen">
    <rosparam file="$(find espeleo_2d_exploration)/launch/nav/costmap_common_params.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find espeleo_2d_exploration)/launch/nav/costmap_common_params.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find espeleo_2d_exploration)/launch/nav/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find espeleo_2d_exploration)/launch/nav/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find espeleo_2d_exploration)/launch/nav/base_local_planner_params.yaml" command="load" />
    <!--<remap from="cmd_vel" to="/bsc_jackal/nav_vel"/>-->
  </node>
  <!--<node pkg="espeleo_control" name="espeleo_control" type="vec_field_control_action.py" args="" output="screen">
    <rosparam command="load" file="$(find espeleo_control)/config/control_params.yaml"/>
  </node>-->
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find espeleo_2d_exploration)/rviz/config_exploration.rviz" />
    <node pkg="espeleo_2d_exploration" type="frontier_publisher_node" name="frontier_publisher_node"/>
    <node pkg="espeleo_2d_exploration" type="distance_measurer" name="distance_measurer" output="screen"/>
    <node pkg="espeleo_2d_exploration" type="espeleo_2d_exploration_node" name="espeleo_2d_exploration_node" output="screen">
     <rosparam command="load" file="$(find espeleo_2d_exploration)/config/config.yaml" />
    </node>
</launch>
